benchmarks/benchmark/tools/profile-generator/main.tf

```diff
@@ -0,0 +1,81 @@
+/**
+ * Copyright 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+provider "kubernetes" {
+  config_path = (
+    var.credentials_config.kubeconfig == null
+    ? null
+    : pathexpand(var.credentials_config.kubeconfig.path)
+  )
+  config_context = try(
+    var.credentials_config.kubeconfig.context, null
+  )
+  host = (
+    var.credentials_config.fleet_host == null
+    ? null
+    : var.credentials_config.fleet_host
+  )
+  token = try(data.google_client_config.identity.0.access_token, null)
+}
+
+data "google_client_config" "identity" {
+  count = var.credentials_config.fleet_host != null ? 1 : 0
+}
+
+resource "google_project_service" "cloudbuild" {
+  count   = var.build_latency_profile_generator_image ? 1 : 0
+  project = var.project_id
+  service = "cloudbuild.googleapis.com"
+
+  timeouts {
+    create = "30m"
+    update = "40m"
+  }
+
+  disable_on_destroy = false
+}
+
+#  ----- Manual Benchmarking -----
+
+module "profile-generator" {
+  count = var.targets.manual != null ? 1 : 0
+  source = "../latency-profile"
```

Comment:
@Bslabe123 @achandrasekar wondering if we should organize this so that it's profile-generator top level, and the latency-profile engine folder within it, eg:

profile-generator/
     latency-profile/
     
This makes it clear via directory level that this is all one tool (eg. akin to how locust and dlio have their own top level file to represent the tooling)

Reply:
Added for this plus deduplicating the image building logic

Reply:
that lgtm, ty ty!